File: app/channels/appearance_channel.rb
class AppearanceChannel < ApplicationCable::Channel
  def subscribed
    Rails.logger.debug "#{current_user.surname} (#{current_user.id}) subscribed"
    stream_for Retrospective.find(params[:retrospective_id])
  end

  def unsubscribed
    Rails.logger.debug "#{current_user.surname} (#{current_user.id}) unsubscribed"
  end

  def receive(data)
    return unless data.fetch('body').in?(['Hello', 'Bonjour'])

    retrospective = Retrospective.find(current_user.retrospective_id)
    Rails.logger.debug "Broadcasting message from #{current_user.surname} (#{current_user.id})"
    AppearanceChannel.broadcast_to(retrospective, body: "#{current_user.surname} says #{data['body']}")
  end
end

File: app/javascript/channels/appearanceChannel.js
import consumer from "./consumer"

export const join = (retrospectiveId) => {
  const appearanceChannel = consumer.subscriptions.create({ channel: 'AppearanceChannel', retrospective_id: retrospectiveId }, {
    connected() {
      console.log('You are connected to the appearance channel!')
      appearanceChannel.send({ body: 'Hello' })
    },
    disconnected() {
      console.log('You were disconnected from the appearance channel!')
    },
    received(data) {
      if (data.new_participant) {
        console.log('New participant', data.new_participant)
      } else if (data.body) {
        console.log(data.body)
      }
    },
  })

  return appearanceChannel
}
